lib/furoshiki/base_app.rb

```diff
@@ -182,13 +199,26 @@ def inject_jar
     def after_built
     end
 
-    def after_moved
-    end
-
-    def move_to_package_dir(path)
-      dest = package_dir.join(app_name)
-      dest.rmtree if dest.exist?
-      mv path.to_s, dest
+    def create_archive(source_path)
+      dest = package_dir.join(archive_name)
+      File.open(dest, "wb") do |file|
+        Zlib::GzipWriter.wrap(file) do |gz|
+          Gem::Package::TarWriter.new(gz) do |tar|
+            tmp_files.each do |source_item|
+              dest_item = source_item.sub(source_path.to_s, app_name)
```

Comment:
If I see this correctly we only need `tar` and `source_item` within the block. Imo we could get all the File, Zlib and Tar  stuff into its own method (maybe also the tmp_files iteration) as the whole wrapping of the file doesn't matter much for what happens inside this, right? Then the method would be much shorter and not so indented :)